packages/vxrn/src/plugins/expoManifestRequestHandlerPlugin.ts

```diff
@@ -0,0 +1,101 @@
+import module from 'node:module'
+import { TLSSocket } from 'node:tls'
+
+import type { Plugin } from 'vite'
+
+type ExpoManifestRequestHandlerPluginConfig = {
+  /** The root of the Expo project. */
+  projectRoot: string
+  // Can support more [options](https://github.com/expo/expo/blob/sdk-50/packages/%40expo/cli/src/start/server/middleware/ManifestMiddleware.ts#L113-L121) in the future.
+}
+
+/**
+ * Let the Vite dev server support handling [Expo Manifest Request](https://github.com/expo/expo/blob/sdk-50/docs/pages/archive/technical-specs/expo-updates-0.mdx#manifest-request), which is required for Expo Go to work.
+ */
+export function expoManifestRequestHandlerPlugin({
+  projectRoot,
+}: ExpoManifestRequestHandlerPluginConfig): Plugin {
+  return {
+    name: 'vxrn:expo-manifest-request-handler',
+
+    configureServer(server) {
+      // Add a middleware to Vite's internal Connect server to handle the Expo Manifest Request.
+      server.middlewares.use(async (req, res, next) => {
+        if (!req.headers['expo-platform']) {
+          // Not an Expo manifest request, skip this middleware and proceed.
+          return next()
+        }
+
+        const protocol = req.socket instanceof TLSSocket && req.socket.encrypted ? 'https' : 'http'
+        const host = `${req.headers['x-forwarded-host']}` || req.headers.host
+
+        // Try to dynamically import the internal Expo manifest handler from expo packages installed in the user's project.
+        let ExpoGoManifestHandlerMiddleware
+        let expoGoManifestHandlerMiddlewareImportError
+        try {
+          // Import Expo from the user's project instead of from where vxrn is installed, since vxrn may be installed globally or at the root workspace.
+          const require = module.createRequire(projectRoot)
+          const importPath = require.resolve(
+            '@expo/cli/build/src/start/server/middleware/ExpoGoManifestHandlerMiddleware.js',
+            { paths: [projectRoot] }
+          )
```

Comment:
Saw that [`import-meta-resolve`](https://www.npmjs.com/package/import-meta-resolve) is being used elsewhere, maybe we can try to use it here too.